lib/restaurant.rb

```diff
@@ -0,0 +1,35 @@
+class Restaurant
+  def view_menu
+    display.view_menu(menu)
+  end
+
+  def add(meal)
+    choice = meal.downcase.to_sym
+    return display.invalid(meal) unless menu.include?(choice)
+
+    order << menu.slice(choice)
+    display.confirm_added(meal)
+  end
+
+  def view_order
+    display.view_order(order)
+  end
+
+  def place_order
+    puts view_order
+    display.confirm_order
+    order.clear
+    customer_service.confirmation_text
+  end
+
+  private
+
+  attr_reader :menu, :order, :display, :customer_service
+
+  def initialize(args = {})
```

Comment:
not 100% sure what's going on here - is this saying that the argument will be a hash with  a customer service object, a display object, and a menu and meals object stored into those keys, which will then be stored to these instance variables? Could you not just have these as 3 arguments? I suppose I'm not quite following where the hash comes in. Also, do you want to be creating a new menu each time? Again, this introduces the possibility of the pizza menu getting added by mistake, when what you'd really want is to be adding a specific menu that a restaurant had created. Though I get that for the purposes of this challenge and testing it it equates to the same thing.

Reply:
I used a hash because with 3 arguments to initialize, there's a serious risk of forgetting arguments or giving them in the wrong order. the hash solves that issue. the conditional assignment `|| =` means any argument left out becomes the default value. It's probably something that I could change in the design on my app. Restaurant is reliant on 3 classes so maybe a change in design would mean this set up isn't necessary and the app would be more flexible